llvm/cmake/modules/GetHostTriple.cmake

```diff
@@ -42,7 +42,21 @@ function( get_host_triple var )
       set( value "powerpc-ibm-aix" )
     endif()
   else( MSVC )
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS)
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS AND CMAKE_C_COMPILER_ID MATCHES "Clang" )
+      if( CMAKE_C_COMPILER_ARCHITECTURE_ID MATCHES "ARM64.*" )
+        set( value "aarch64-pc-windows-msvc" )
+      elseif( CMAKE_C_COMPILER_ARCHITECTURE_ID MATCHES "ARM.*" )
+        set( value "armv7-pc-windows-msvc" )
+      elseif( CMAKE_C_COMPILER_ARCHITECTURE_ID STREQUAL "x64" )
+        set( value "x86_64-pc-windows-msvc" )
+      elseif( CMAKE_C_COMPILER_ARCHITECTURE_ID STREQUAL "X86" )
+        set( value "i686-pc-windows-msvc" )
+      elseif( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+        set( value "x86_64-pc-windows-msvc" )
+      else()
+        set( value "i686-pc-windows-msvc" )
```

Comment:
Should we assume x86/x64 when `CMAKE_C_COMPILER_ARCHITECTURE_ID` isn't `x86` or `x64`? Can we report an error instead?